client/src/components/TaskFilterSelector/TaskFilterSelector.js

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import styles from "./TaskFilterSelector.module.css";
+import { PropTypes } from "prop-types";
+
+const FilterButton = ({ filterName, currentFilter, onFilterChanged }) => {
+  return (
+    <button
+      className={
+        filterName === currentFilter
+          ? styles.selectedFilter
+          : styles.filterButton
+      }
+      onClick={() => onFilterChanged(filterName)}
+    >
+      {filterName}
+    </button>
+  );
+};
+
+FilterButton.propTypes = {
+  filterName: PropTypes.string.isRequired,
+  currentFilter: PropTypes.string.isRequired,
+  onFilterChanged: PropTypes.func.isRequired,
+};
+
+export const TaskFilterSelector = ({
+  filterSet,
+  currentFilter,
+  onFilterChanged,
+}) => {
+  const filterButtons = filterSet.map((filter, ix) => (
```

Comment:
I would also just rename this to `filterName` instead of filter. (Since it's the filter name from the filterSet

Reply:
Renamed `filter` to  `filterName`.